### PR TITLE
Update codeowners file

### LINKED ISSUE #14054: [Meta/PM] Update CODEOWNERS
```[tasklist]
## Todo
- [x] Update rules for `sp-artithmetic` and more
- [x] Old users should be replaced
- [x] Make frame-coders own `/frame/`
```

### PR DESCRIPTION
**Pull request by juangirini**
Updates the CODEOWNERS file

Fixes #14054

### CODE REVIEW

File: docs/CODEOWNERS [hunk 1]
```diff
@@ -63,10 +61,4 @@
 /frame/staking/ @paritytech/staking-core
 /frame/nomination-pools/ @paritytech/staking-core
 /frame/elections-phragmen/ @paritytech/staking-core
```
Comment (ggwpez) [inline docs/CODEOWNERS:63]: Could you maybe sort them? Then it is easier to spot missing lie `election-provider-multi-phase`.

File: docs/CODEOWNERS [hunk 2]
```diff
@@ -38,6 +35,9 @@
 /client/sysinfo/ @koute
 /client/tracing/ @koute
 
+# FRAME
+/frame/ @paritytech/frame-coders
+
```
Comment (ggwpez) [inline docs/CODEOWNERS:40]: Please add `paritytech/frame-coders` and `cheme` for `/frame/state-trie-migration/`.

**Review by ggwpez — COMMENTED** [review]
:+1:

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
Hm i have on idea how to approve on behalf of the team or not :laughing:

**Review by athei — APPROVED** [review]

File: docs/CODEOWNERS [hunk 3]
```diff
@@ -19,54 +19,48 @@
...
-/primitives/npos-elections/ @paritytech/staking-core
-
-# Fixed point arithmetic
-/primitives/sp-arithmetic/ @kianenigma
-
-# Transaction weight stuff
-/frame/support/src/weights.rs @shawntabrizi
+/frame/nomination-pools/ @paritytech/staking-core
+/frame/staking/ @paritytech/staking-core
+/primitives/npos-elections/ @paritytech/staking-core
```
Comment (kianenigma) [inline docs/CODEOWNERS:66]:
```suggestion
/primitives/npos-elections/ @paritytech/staking-core

```

**Review by kianenigma — APPROVED** [review]

**Comment by kianenigma** [comment]
Cool, so this is only useful if @paritytech/frame-coders actually look at their notifications though, which I hope they do 😜

**Comment by kianenigma** [comment]
I wonder if some pallets that are handled by @joepetrowski and team can also have dedicated owners; eg uniques and NFT to @jsidorenko?

**Comment by juangirini** [comment]
bot merge

**Pull request merged**